docs/source/api_reference/data_format.rst

```diff
@@ -0,0 +1,273 @@
+.. _data_format:
+
+File Format Specifications
+==========================
+
+.. toctree::
+    :maxdepth: 1
+    :hidden: false
+
+    Overview
+    Introduction
+    Description
+    Instructions
+    Illustration
+
+Overview
+--------
+This document has the following content:
+
+- Introduction: What is a ``.ts`` file, why and when to use it.
+- Description: What are the individual components of a ``.ts`` file.
+- Instructions: How to create your own ``.ts`` file.
+- Illustrations: A running example to tie up above sections.
+
+Introduction
+------------
+This document formalizes string identifiers used in ``.ts`` file format.
+Encoded in ``utf-8``, ``.ts`` files stores time-series dataset and its corresponding metadata and can be
+opened via any basic editor like notepad for visual inspection.
+String identifiers refer to strings beginning with ``@`` in the file.
+
+``.ts`` files contains information blocks in the following order:
+
+1. All the comments to describe the dataset. (begins with ``#`` in the file)
+2. All the String Identifiers to give information about the metadata (begins with ``@`` in the file).
+3. Actual dataset.
+
+See `Basic Motion.ts`_ for a concrete example.
```

Comment:
I would link the file, but perhaps giving extracts directly here is better.

I.e., show the file with some annotation on what the blocks are. Then, ensure to link to the sections that give detail descriptions.

Reply:
Are you suggesting adding the contents from that file into the doc? I think that will needlessly increase the size, especially since we already have an illustration. I could just add a cross-ref to the illustration.

Reply:
I would just show extracts, not the entire file. Or move the reference to later, so the reaser is not lead to another file too early.